lib/commons/dom/is-skip-link.js

```diff
@@ -13,22 +16,41 @@ const isInternalLinkRegex = /^\/?#[^/!]/;
  * @return {Boolean}
  */
 function isSkipLink(element) {
-  if (!isInternalLinkRegex.test(element.getAttribute('href'))) {
-    return false;
-  }
-
   let firstPageLink;
   if (typeof cache.get('firstPageLink') !== 'undefined') {
     firstPageLink = cache.get('firstPageLink');
   } else {
-    // define a skip link as any anchor element whose href starts with `#...`
-    // and which precedes the first anchor element whose href doesn't start
-    // with  `#...` (that is, a link to a page)
+    // define a skip link as any anchor element whose resolved href
+    // resolves to the current page and uses a fragment identifier (#)
+    // and which precedes the first anchor element whose resolved href
+    // does not resolve to the current page or that doesn't use a
+    // fragment identifier
+    const currentPage = window.location.origin + window.location.pathname;
     firstPageLink = querySelectorAll(
-      // TODO: es-module-_tree
       axe._tree,
-      'a:not([href^="#"]):not([href^="/#"]):not([href^="javascript"])'
-    )[0];
+      'a[href]:not([href^="javascript"])'
```

Comment:
Should that be this: 

```suggestion
      'a[href]:not([href^="javascript:"])'
```

It's unlikely, but no reason to think someone can't just have `<a href="javascript/home.php">` or whatever.